### PR TITLE
Fix for Windows SearchBar MaxLength > 0 not working properly.

### LINKED ISSUE #5669: Windows SearchBar MaxLength > 0 not working properly
### Description

On Windows, the platform control AutoSuggestBox does not have a built-in property for MaxLength

Therefore, we are having some issues with getting MaxLength to fully work for SearchBar on Windows. For MaxLength > 0, any preassigned text gets truncated by our substring logic - however, it does not limit the searchbar itself and text greater than the assigned maxlength can still be inputted.

Creating an attached property (as was done with this control for IsReadOnly did not make a difference.

### Steps to Reproduce

1. Create a File > New > .NET MAUI app
2. Add `<SearchBar Text="1 SearchBar text" MaxLength="1" />` to the MainPage.xaml
3. Deploy app to Windows machine
4. Notice the SearchBar text only shows the first character "1"
5. Click into the SearchBar, type more, and notice that it is possible when it shouldn't be

### Version with bug

Release Candidate 1

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

all

### Did you find any workaround?

A potential workaround is calling UpdateMaxLength() every time the text is updated
However, this is not ideal and is very laggy/glitchy, having the appearance of text being deleted after the fact

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BagavathiPerumal**
### Root cause

WinUI AutoSuggestbox does not have MaxLength property in native, SearchBar.MaxLength is not updated to native property.

### Description of Issue Fix

Modified SearchBar by taking the AutoSuggestbox child, a textbox control in UpdateMaxLength() and updated SearchBar.MaxLength to native textbox.MaxLength.

Tested the behavior in the following platforms.
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Issues Fixed

Fixes #5669

### Output

| Before Issue Fix | After Issue Fix |
|----------|----------|
| <video width="270" height="600"  src="https://github.com/user-attachments/assets/3f4fa6f2-b30e-4e73-b8c9-a7a79976cd5f"> | <video width="270" height="600"  src="https://github.com/user-attachments/assets/cd55d115-f6dd-4c67-9f9d-0b1f37f2cb8b"> |

### CODE REVIEW

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 1]
```diff
@@ -107,11 +107,14 @@ public static void UpdateVerticalTextAlignment(this AutoSuggestBox platformContr
...
 		{
+			var textbox = FindChildOfType<TextBox>(platformControl);
+
 			var maxLength = searchBar.MaxLength;
 
 			if (maxLength == -1)
 				maxLength = int.MaxValue;
-
+			if (textbox != null)
+				textbox!.MaxLength = searchBar.MaxLength;
```
[the remark this answers is not in the document]
Comment (BagavathiPerumal) [reply src/Core/src/Platform/Windows/SearchBarExtensions.cs]: @MartyIX, I have modified changes. Could you please validate it once and let me know if there are any further concerns.

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 2]
```diff
@@ -112,6 +112,12 @@ public static void UpdateMaxLength(this AutoSuggestBox platformControl, ISearchB
 			if (maxLength == -1)
 				maxLength = int.MaxValue;
 
+			var child = platformControl.GetChildren<TextBox>();
+			if (child is not null && child.Count() > 0)
+			{
+				child.FirstOrDefault()!.MaxLength = searchBar.MaxLength;
+			}
```
[the remark this answers is not in the document]
Comment (BagavathiPerumal) [reply src/Core/src/Platform/Windows/SearchBarExtensions.cs]: @MartyIX, I have modified changes based on your suggestion. Could you please validate it once and let me know if there are any further concerns.

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 3]
```diff
@@ -112,6 +112,17 @@ public static void UpdateMaxLength(this AutoSuggestBox platformControl, ISearchB
 			if (maxLength == -1)
 				maxLength = int.MaxValue;
 
+			var child = platformControl.GetChildren<TextBox>();
+
+			if (child is not null && child.Count() > 0)
```
[the remark this answers is not in the document]
Comment (BagavathiPerumal) [reply src/Core/src/Platform/Windows/SearchBarExtensions.cs]: @MartyIX, I have modified changes. Could you please validate it once and let me know if there are any further concerns.

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 4]
```diff
@@ -112,6 +112,17 @@ public static void UpdateMaxLength(this AutoSuggestBox platformControl, ISearchB
 			if (maxLength == -1)
 				maxLength = int.MaxValue;
 
+			var child = platformControl.GetChildren<TextBox>();
```
[the remark this answers is not in the document]
Comment (BagavathiPerumal) [reply src/Core/src/Platform/Windows/SearchBarExtensions.cs]: @MartyIX, I have made changes based on your concerns. Please validate and let me know if there are any issues.

**Review by rmarinho — DISMISSED** [review]
I think it's missing some using the builds is failing

```
D:\a\1\s\src\Core\src\Platform\Windows\SearchBarExtensions.cs(118,28): error CS1061: 'IEnumerable<TextBox?>' does not contain a definition for 'FirstOrDefault' and no accessible extension method 'FirstOrDefault' accepting a first argument of type 'IEnumerable<TextBox?>' could be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-windows10.0.20348.0]
```

**Comment by BagavathiPerumal** [comment]
> I think it's missing some using the builds is failing
> 
> ```
> D:\a\1\s\src\Core\src\Platform\Windows\SearchBarExtensions.cs(118,28): error CS1061: 'IEnumerable<TextBox?>' does not contain a definition for 'FirstOrDefault' and no accessible extension method 'FirstOrDefault' accepting a first argument of type 'IEnumerable<TextBox?>' could be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\src\Core\src\Core.csproj::TargetFramework=net8.0-windows10.0.20348.0]
> ```

@rmarinho, I have modified changes for this issue. Could you please validate it once and let me know if there are any further concerns.

**Comment by rmarinho** [comment]
Can we also rebase on main? we made some changes on the pipeline agents and we need the pr to pick them up. 

Thanks

**Comment by BagavathiPerumal** [comment]
> Can we also rebase on main? we made some changes on the pipeline agents and we need the pr to pick them up.
> 
> Thanks

@rmarinho, I have rebased this branch successfully.

**Review by PureWeen — CHANGES_REQUESTED** [review]
SearchBarMaxLength failing

**Comment by BagavathiPerumal** [comment]
> SearchBarMaxLength failing

@PureWeen, I have updated test image for this issue. Could you please validate it once and let me know if there are any further concerns.

File: src/Core/src/Platform/Windows/SearchBarExtensions.cs [hunk 5]
```diff
@@ -112,6 +113,16 @@ public static void UpdateMaxLength(this AutoSuggestBox platformControl, ISearchB
 			if (maxLength == -1)
 				maxLength = int.MaxValue;
 
+			var children = platformControl.GetChildren<TextBox>();
+			if (children is not null)
+			{
+				var textBox = children.FirstOrDefault();
```
Comment (PureWeen) [inline src/Core/src/Platform/Windows/SearchBarExtensions.cs:119]: By convention we stay away from using Linq

Can you remove the use of linq here?

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by Foda — DISMISSED** [review]
LGTM, just please address LINQ comment

**Comment by BagavathiPerumal** [comment]
> LGTM, just please address LINQ comment

@Foda, I have updated the code based on your suggestion. Could you please review it and let me know if you have any further concerns.

**Review by PureWeen — DISMISSED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**